File: Game.cpp
#include "chess_gui.h"

Game::Game(std::string white_name, std::string black_name)
    : _black_player(black_name, _chessboard, Color::Black),
      _white_player(white_name, _chessboard, Color::White),
      _white_move(true)
{
}

bool Game::enter_move(std::string move)
{
    bool successful;
    try
    {
        if (_white_move)
        {
            _white_player.make_move(move);
        }
        else
        {
            _black_player.make_move(move);
        }

        _white_move = !_white_move;
    }
    catch (std::logic_error ex)
    {
        std::cout << ex.what() << "\n";
    }

    // TODO store moves here
    return true;
}

void Game::print_board()
{
    _chessboard.print_board();
}

File: src/com/keshane/Paw/Game.java
package com.keshane.Paw;

import java.text.ParseException;

public class Game {
    private final Board board;

    private final Player whitePlayer;
    private final Player blackPlayer;

    private Color currentTurn;

    private final Move.History history;

    Game(String whiteName, String blackName, Board.Configuration configuration) {
        board = new Board(configuration);
        whitePlayer = new Player(whiteName, Color.WHITE, new Player.WhiteStandardLogic());
        blackPlayer = new Player(blackName, Color.BLACK, new Player.BlackStandardLogic());
        currentTurn = Color.WHITE;
        history = new Move.History();
    }

    public void makeMove(String notation) throws ParseException, NoSuchMoveException,
            AmbiguousNotationException, KingInCheckException {
        Move move;
        if (currentTurn == Color.WHITE) {
            move = whitePlayer.makeMove(notation, history, board);
            currentTurn = Color.BLACK;
        }
        else {
            move = blackPlayer.makeMove(notation, history, board);
            currentTurn = Color.WHITE;
        }

        history.add(move);
    }

    public void printBoard(Color perspective) {
        System.out.println(board.toString(perspective));
    }

    public void printHistory() {
        System.out.println(history.toString());
    }

    public Color getCurrentTurn() {
        return currentTurn;
    }



}
